### PR TITLE
When revwalking only get the commits from the first parent

### PR DESCRIPTION
**Pull request by krlvi**
Also adding an option for getting commits from all parents

### CODE REVIEW

**Review by Caleb-T-Owens — APPROVED** [review]
I'm stunned this affected any tests

Edit: I realize that this is actually a draft

**Comment by Caleb-T-Owens** [comment]
CC @Byron . Thought it would make sense to notify you about this change

**Comment by krlvi** [comment]
> I realize that this is actually a draft

this time - not a draft :D

File: crates/gitbutler-branch-actions/tests/extra/mod.rs [hunk 1]
```diff
@@ -1009,7 +1009,9 @@ fn merge_vbranch_upstream_conflict() -> Result<()> {
     );
 
     assert_eq!(branch1.files.len(), 0);
-    assert_eq!(branch1.commits.len(), 4); // Local commit + Remote commit + Merge commit + Conflicted uncommited changes
+    assert_eq!(branch1.commits.len(), 3); // Local commit + Remote commit + Merge commit + Conflicted uncommited changes
```
Comment (Caleb-T-Owens) [inline crates/gitbutler-branch-actions/tests/extra/mod.rs:1012]: Can we update the comment to explain why remote commit is not listed there

**Comment by Caleb-T-Owens** [comment]
Other than the one comment, looks even better to me

**Pull request merged**